### PR TITLE
Add support for Inovelli NZW31 switch config, add additional device IDs

### PR DESCRIPTION
**Pull request by jdeglopper**
Based on Inovelli-provided SmartThings device handler, and tested against a hardware NZW31 switch.

### CODE REVIEW

**Comment by vondit** [comment]
Great minds?  lol
https://github.com/OpenZWave/open-zwave/pull/1469

One quick question:  Are you sure the NZW3xT is also Scene?  i think the T simply denotes to the physical toggle switch versus rocker switch.  I'm waiting on my order to arrive to fully confirm, but I believe there are 4 different Inovelli switches.  Rocker and Toggle On/Off and Rocker and Toggle Dimmer.  Each one of those would have a Scene / Non-Scene version.  I'm just waiting to get mine to confirm the device IDs

**Comment by jdeglopper** [comment]
Parallel evolution, clearly.  The website shows 6 varieties of switches, not 8 - both of the toggle models there have scene support. https://inovelli.com/zwave-switches/  

I pulled the device IDs from their SmartThings device handler.

**Comment by vondit** [comment]
Ha.  Good eye.  Totally missed that...

Probably the wrong place for this, but how are you liking the Inovelli switches?  I stumbled upon them by accident and they've become my go-to.

**Comment by nechry** [comment]
@jdeglopper  can you rebase your PR on latest version with #1469 PR

**Comment by InovelliUSA** [comment]
Hey all -- Just popping in to confirm the following:

> One quick question: Are you sure the NZW3xT is also Scene? i think the T simply denotes to the physical toggle switch versus rocker switch. I'm waiting on my order to arrive to fully confirm, but I believe there are 4 different Inovelli switches. Rocker and Toggle On/Off and Rocker and Toggle Dimmer. Each one of those would have a Scene / Non-Scene version. I'm just waiting to get mine to confirm the device IDs

Yes, the NZW3xT's both handle scene support. But you're also correct in your assumption that the, "T" stands for Toggle :)

Here are the following models of switches:

- NZW30 = On/Off (Non-Scene) / Paddle
- NZW31 = Dimming (Non-Scene) / Paddle
- NZW30S = On/Off (Scene) / Paddle
- NZW31S = Dimming (Scene) / Paddle
- NZW30T = On/Off (Scene) / Toggle
- NZW31T = Dimming (Scene) / Toggle

There was a silkscreen error on the first round of paddle/scene switches in that the manufacturer forgot to put the "S" after the model number.

@vondit -- I'll be shipping out the toggles today (03/23/18). 

In regards to the fingerprint/ID's, some of the older switches may have a conflict as we realized later that the manufacturer put the Zooz Manufacturing ID on them instead of ours. This is why there are so many ID's in the handler (it was quite the nightmare)!

Anyway, thanks so much for the kind words and have an awesome day!

Eric
Founder | Inovelli

File: config/inovelli/nzw31.xml [hunk 1]
```diff
@@ -1,12 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
+<!-- Inovelli Dimmer NZW31 -->
 <Product xmlns="http://code.google.com/p/open-zwave/">
   <!-- Configuration Parameters -->
   <CommandClass id="112">
-    <Value genre="config" index="1" label="Dimming/Brightening Step Level" min="1" max="99" value="1" size="1" type="byte">
-      <Help>Each step of the dimming process will increase or decrease the brightness by this percentage. Increasing this will make the transition time quicker.</Help>
+    <Value genre="config" index="1" label="Dimming/Brightening Step Level" min="1" max="99" units="%" value="1" size="1" type="byte">
```
Comment (brianlow) [inline config/inovelli/nzw31.xml]: On my NZW31 (type `1f01`, id `1f01`), parameter 1 can be set to 0 to disable dimming (works like on/off).

Comment (InovelliUSA) [reply config/inovelli/nzw31.xml]: Hi there, just to confirm. There was an April firmware change that allows for a setting of 0 for parameter 1. This indeed causes the dimmer to instantly turn on/off like a typically switch. 

Eric M
VP Technology | Inovelli

**Pull request merged**